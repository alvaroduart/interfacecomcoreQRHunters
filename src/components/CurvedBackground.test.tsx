import React from 'react';
import { render } from '@testing-library/react-native';
import CurvedBackground from './CurvedBackground';

describe('CurvedBackground', () => {
  it('renders with default props', () => {
    const { toJSON } = render(<CurvedBackground />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders with custom color, height, borderRadius, and position', () => {
    const { toJSON } = render(
      <CurvedBackground
        color="#123456"
        height={200}
        borderRadius={50}
        position="top"
      />
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders with position bottom', () => {
    const { toJSON } = render(
      <CurvedBackground position="bottom" />
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
