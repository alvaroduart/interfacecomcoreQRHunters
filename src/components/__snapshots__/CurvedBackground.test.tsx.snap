// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurvedBackground renders with custom color, height, borderRadius, and position 1`] = `
<View
  style={
    [
      {
        "left": 0,
        "position": "absolute",
        "right": 0,
        "zIndex": 1,
      },
      {
        "backgroundColor": "#123456",
        "borderBottomLeftRadius": 50,
        "borderBottomRightRadius": 50,
        "height": 200,
        "top": 0,
      },
      {},
    ]
  }
/>
`;

exports[`CurvedBackground renders with default props 1`] = `
<View
  style={
    [
      {
        "left": 0,
        "position": "absolute",
        "right": 0,
        "zIndex": 1,
      },
      {
        "backgroundColor": "#4169e1",
        "borderTopLeftRadius": 130,
        "borderTopRightRadius": 130,
        "bottom": 0,
        "height": "40%",
      },
      {},
    ]
  }
/>
`;

exports[`CurvedBackground renders with position bottom 1`] = `
<View
  style={
    [
      {
        "left": 0,
        "position": "absolute",
        "right": 0,
        "zIndex": 1,
      },
      {
        "backgroundColor": "#4169e1",
        "borderTopLeftRadius": 130,
        "borderTopRightRadius": 130,
        "bottom": 0,
        "height": "40%",
      },
      {},
    ]
  }
/>
`;
